segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -187,4 +198,28 @@ static void delete(long ledgerId, BookKeeper bookKeeper) throws DurableDataLogEx
 
         return result;
     }
+
+    /**
+     * Gets the {@link #PROPERTY_LOG_ID} value from the given {@link LedgerHandle}, as stored in its custom metadata.
+     *
+     * @param handle The {@link LedgerHandle} to query.
+     * @return The Log Id stored in {@link #PROPERTY_LOG_ID}, or {@link #NO_LOG_ID} if not defined (i.e., due to an upgrade
+     * from a version that did not store this information).
+     */
+    static int getBookKeeperLogId(LedgerHandle handle) {
+        byte[] logIdSerialized = handle.getCustomMetadata().getOrDefault(PROPERTY_LOG_ID, null);
+        return logIdSerialized == null || logIdSerialized.length < Integer.BYTES ? NO_LOG_ID : BitConverter.readInt(logIdSerialized, 0);
+    }
+
+    /**
+     * Creates a new {@link Map} that can be set as a Ledger's Custom Metadata.
+     *
+     * @param logId The {@link BookKeeperLog} Id to encode.
+     * @return An immutable {@link Map} containing the encoded Ledger's Custom Metadata.
+     */
+    private static Map<String, byte[]> createLedgerCustomMetadata(int logId) {
+        byte[] logIdSerialized = new byte[Integer.BYTES];
+        BitConverter.writeInt(logIdSerialized, 0, logId);
```

Comment:
What about writing this as a string?
The tools I know that manage custom metadata actually assume that the values of the map are utf8 encoded strings.
It is not a big waste to use strings because this info is read/written out of the hot read/write paths

Reply:
Interesting idea. Let me see what I can do there. That would also make it easier to edit the data directly in ZK, without the use of this tool.